src/modules/_networking/local_network_gateways/local_network_gateway.tf

```diff
@@ -0,0 +1,7 @@
+resource "azurerm_local_network_gateway" "main" {
+  name                = var.settings.name
+  location            = local.location
+  resource_group_name = local.resource_group_name
+  gateway_address     = var.settings.gateway_address
```

Comment:
Add an option for [gateway_fqdn](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/local_network_gateway.html#gateway_fqdn-1) as well. Make sure they're both optional.
No need for validation, we can trust the provider to check either is provided.